projects/knora-ui/src/lib/viewer/values/time-value/time-value.component.ts

```diff
@@ -0,0 +1,198 @@
+import { Component, OnInit, OnChanges, OnDestroy, ViewChild, Input, Inject, SimpleChanges, NgZone } from '@angular/core';
+import { CdkTextareaAutosize } from '@angular/cdk/text-field';
+import { take } from 'rxjs/operators';
+import { TimeInputComponent, DateTime } from './time-input/time-input.component';
+import { ReadTimeValue, CreateTimeValue, UpdateTimeValue, KnoraDate } from '@knora/api';
+import { BaseValueComponent } from '..';
+import { FormControl, FormGroup, FormBuilder, ValidatorFn, AbstractControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { IntervalErrorStateMatcher } from '../interval-value/interval-value.component';
+import { DatePipe } from '@angular/common';
+import * as _ from 'lodash';
+import { CustomRegex } from '../custom-regex';
+
+@Component({
+  selector: 'kui-time-value',
+  templateUrl: './time-value.component.html',
+  styleUrls: ['./time-value.component.scss']
+})
+export class TimeValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @ViewChild('timeInput', {static: false}) timeInputComponent: TimeInputComponent;
+  @ViewChild('autosize', {static: false}) autosize: CdkTextareaAutosize;
+
+  @Input() displayValue?: ReadTimeValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  date: string;
+  time: string;
+  initKnoraDate: KnoraDate; // used to display the initial value which will be an instance of KnoraDate
```

Comment:
Couldn't these be variables defined in a method's local scope? As far as I see writing and reading happens in the same place for `initKnoraDate`, `convertedControlDate`, and `updateDate`, respectively.

Reply:
Removed in 046010e